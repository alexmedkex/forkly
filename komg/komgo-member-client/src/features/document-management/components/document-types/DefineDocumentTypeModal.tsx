import * as React from 'react'
import { Form, Modal, Button, Input, Dropdown } from 'semantic-ui-react'
import { Category } from '../../store/types'

export interface Props {
  visible: boolean
  title: string
  categories: Category[]
  predefinedData: { category: string; name: string; id?: string }
  toggleVisible(): void
  onCreateSuccess(cat: string, name: string): void
  onEditSuccess(cat: string, name: string, id: string): void
}

interface State {
  category: string
  name: string
}

interface CategoryRepresentation {
  key: string
}

const formatCategoriesToSelect = (categories: Category[]): CategoryRepresentation[] => {
  return categories.map((cat: Category) => {
    return { key: cat.id, text: cat.name, value: cat.id }
  })
}

class DefineDocumentTypeModal extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props)
  }

  handleSubmit = () => {
    if (this.props.predefinedData.id) {
      this.props.onEditSuccess(
        this.state.category ? this.state.category : this.props.predefinedData.category,
        this.state.name,
        this.props.predefinedData.id
      )
    } else {
      this.props.onCreateSuccess(this.state.category, this.state.name)
    }
  }

  render() {
    return (
      <Modal open={this.props.visible} size="mini">
        <Modal.Header>{this.props.title}</Modal.Header>
        <Modal.Content>
          <Modal.Description>
            <Form onSubmit={this.handleSubmit} id="submit-form">
              <Form.Field>
                <label>Category</label>
                <Dropdown
                  placeholder="Select category"
                  defaultValue={
                    this.props.predefinedData.category !== '' ? this.props.predefinedData.category : undefined
                  }
                  options={formatCategoriesToSelect(this.props.categories)}
                  onChange={this.setSelectedCategoryId}
                />
              </Form.Field>
              <Form.Field>
                <label>Name</label>
                <Input
                  placeholder="My new doctype"
                  defaultValue={this.props.predefinedData.name !== '' ? this.props.predefinedData.name : undefined}
                  onChange={(e: any) => {
                    this.setState({ name: e.target.value })
                  }}
                  size="mini"
                />
              </Form.Field>
            </Form>
          </Modal.Description>
        </Modal.Content>
        <Modal.Actions>
          <Button className="ui negative button" role="button" onClick={this.props.toggleVisible}>
            Cancel
          </Button>
          <Button primary={true} content="Submit" form="submit-form" />
        </Modal.Actions>
      </Modal>
    )
  }

  private setSelectedCategoryId = (e: any, { value }: { value: string }) => {
    this.setState({ category: value })
  }
}

export default DefineDocumentTypeModal
