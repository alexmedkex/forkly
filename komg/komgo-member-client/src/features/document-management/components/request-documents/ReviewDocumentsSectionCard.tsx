import * as React from 'react'
import { SectionCard } from './SectionCard'
import styled from 'styled-components'
import { Request, DocumentType, Document, Category } from '../../store'
import { List } from 'semantic-ui-react'
import { blueGrey, violetBlue } from '../../../../styles/colors'
import { TileWithItems } from './Tile'
import { DownloadButton } from '../documents/DownloadButton'

import { sortCategories, sortDocumentTypes } from '../../utils/sortingHelper'
import _ from 'lodash'
import SpanAsLink from '../../../../components/span-as-link/SpanAsLink'
import ViewMultipleDocuments from './ViewMultipleDocuments'
import { AttachDocumentsDropdown } from './requirements-section/AttachDocumentsDropdown'
import { IAttachedDocument } from '../../containers/DocumentRequestContainer'
import { SPACES } from '@komgo/ui-components'
import AlreadySentTabSubsectionCard from './AlreadySentTabSubsectionCard'

export interface Props {
  documentRequest: Request
  documentsByType: Map<string, Document[]>
  attachedDocuments: Map<string, IAttachedDocument[]>
  downloadedRequestAttachmentForTypes: string[]
  openViewDocument(previewDocumentId: string): void
  onOriginalDocument(doc: Document): void
  onDocumentAttachmentDownload(doc: Document): void
  automatchSelectRequested(docType: DocumentType): void
  addNewDocumentRequested(docType: DocumentType): void
  deleteDocRequested(docId: string): void
  resetLoadedDocument(): void
}

export interface State {
  activePreviewForType: DocumentType
  selectedTab: TABS_NAMES
}

enum TABS_NAMES {
  requested = 'Requested',
  alreadySent = 'Already sent',
  dismissed = 'Dismissed'
}

const title: string = 'REQUEST DETAILS'

export class ReviewDocumentsSectionCard extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props)

    this.state = {
      activePreviewForType: null,
      selectedTab: TABS_NAMES.requested
    }

    this.handleViewDocuments = this.handleViewDocuments.bind(this)
  }

  render() {
    const alreadySentDocs = this.filterAlreadySelectedDocs(
      this.props.documentsByType,
      this.props.documentRequest.sentDocuments
    )
    if (this.props.documentRequest) {
      return (
        <>
          <SectionCard title={title} height={'400px'}>
            {this.renderTabheader(this.getNumberOfAlreadySentDocTypes(alreadySentDocs))}
            {this.renderBody(this.state.selectedTab, alreadySentDocs)}
          </SectionCard>
          {this.renderDocumentsPreview()}
        </>
      )
    } else {
      return null
    }
  }

  private renderBody(tabSelected: TABS_NAMES, alreadySentDocs: Map<string, Document[]>) {
    if (tabSelected === TABS_NAMES.requested) {
      return <StyledListDocTypes className="style-scroll">{this.renderDocuments()}</StyledListDocTypes>
    } else if (tabSelected === TABS_NAMES.alreadySent) {
      return (
        <AlreadySentTabSubsectionCard
          counterpartyId={this.props.documentRequest.companyId}
          alreadySentDocs={alreadySentDocs}
          openViewDocument={this.props.openViewDocument}
        />
      )
    }
  }

  /**
   *
   * @param documentsByType a map with docTypeId as key and array of documents as value. It represents
   * the list of all documents associated to this request we are reviewing
   * @param sentDocuments its a list with all the documents we already sent to our counterparty in this request
   *
   * The result is a map with the same structure as the first parameter, but filtering only the documents
   * we have already sent to the counterparty. So the key is an id of document type and the value is an array
   * of Documents.
   */
  private filterAlreadySelectedDocs(
    documentsByType: Map<string, Document[]>,
    sentDocuments: string[]
  ): Map<string, Document[]> {
    const mapAlreadyShared = new Map<string, Document[]>()
    documentsByType.forEach((value, key) => {
      value.forEach(doc => {
        if (sentDocuments.includes(doc.id)) {
          if (mapAlreadyShared.has(key)) {
            // We have already documents of this docType yet
            mapAlreadyShared.set(key, [...mapAlreadyShared.get(key), doc])
          } else {
            // There is no documents of this docType yet
            mapAlreadyShared.set(key, [doc])
          }
        }
      })
    })
    return mapAlreadyShared
  }

  private renderTabheader(numberOfAlreadySentDocs: number) {
    return (
      <StyledContent>
        <StyledTab>
          <StyledTabItem
            onClick={() => this.changeTab(TABS_NAMES.requested)}
            active={this.state.selectedTab === TABS_NAMES.requested}
          >
            {`${TABS_NAMES.requested} `} <CountWrap>({this.getNumberOfRequestedDocumentTypes()})</CountWrap>
          </StyledTabItem>
          <StyledTabItem
            onClick={() => numberOfAlreadySentDocs > 0 && this.changeTab(TABS_NAMES.alreadySent)}
            active={this.state.selectedTab === TABS_NAMES.alreadySent}
          >
            {`${TABS_NAMES.alreadySent} `} <CountWrap>({numberOfAlreadySentDocs})</CountWrap>
          </StyledTabItem>
        </StyledTab>
      </StyledContent>
    )
  }

  private changeTab(tab: TABS_NAMES) {
    this.setState({ ...this.state, selectedTab: tab })
  }

  private renderDocumentsPreview() {
    const { activePreviewForType } = this.state
    const { documentsByType } = this.props

    if (!activePreviewForType) {
      return null
    }

    const documentsIds = this.getDocumentIds(activePreviewForType)

    const documentsInLibraryForType = documentsByType.has(activePreviewForType.id)
      ? documentsByType.get(activePreviewForType.id)
      : []

    const docsForPreview = documentsIds.filter(docId => {
      const doc = documentsInLibraryForType.find(d => d.id === docId)

      return doc && doc.state === 'REGISTERED'
    })

    return (
      documentsIds.length > 0 && (
        <ViewMultipleDocuments
          documentIds={docsForPreview}
          closeModal={this.handleViewDocuments}
          delete={docId => this.handleDocDelete(docsForPreview, docId)}
        />
      )
    )
  }

  private getDocumentIds(docType?: DocumentType) {
    const { attachedDocuments } = this.props

    if (docType) {
      return attachedDocuments.has(docType.id) ? attachedDocuments.get(docType.id).map(doc => doc.documentId) : null
    }

    return Array.from(attachedDocuments.entries()).reduce(
      (memo, documentsPerType) => [...memo, ...documentsPerType[1].map(d => d.documentId)],
      []
    )
  }

  private renderDocuments() {
    // retrieve all requested document categories, sort them alphabetically and remove duplicates
    const categories: Category[] = this.props.documentRequest.types.map(type => type.category)
    const sortedCategories: string[] = sortCategories(categories).map(category => category.id)
    const uniqSortedCategories = _.uniq(sortedCategories)

    // for each sorted category, sort its requested document types
    let sortedDocumentTypes: DocumentType[] = []
    for (const categoryId of uniqSortedCategories) {
      const typesWithinCategory = this.props.documentRequest.types.filter(
        documentType => documentType.category.id === categoryId
      )
      const sortedTypesWithinCategory = sortDocumentTypes(typesWithinCategory)
      sortedDocumentTypes = sortedDocumentTypes.concat(sortedTypesWithinCategory)
    }

    // render a tile per document type in the correct order
    return sortedDocumentTypes.map(type => this.renderTile(type))
  }

  private handleViewDocuments(docType?: DocumentType) {
    this.setState({
      activePreviewForType: docType
    })
    if (!docType) {
      this.props.resetLoadedDocument()
    }
  }

  private canAttachDocForType(requestAttachment: Document, type: string) {
    const { downloadedRequestAttachmentForTypes = [] } = this.props

    return (
      !requestAttachment ||
      (requestAttachment &&
        requestAttachment.downloadInfo &&
        requestAttachment.downloadInfo.downloadedByUsers.length > 0) ||
      downloadedRequestAttachmentForTypes.includes(type)
    )
  }

  private renderTile(docType: DocumentType) {
    const renderTileItems = () => {
      const docRequestAttachments = this.props.documentRequest.documents.filter(doc => doc.type.id === docType.id)

      // we only assume a single document on the request (i.e, a form)
      const requestAttachment = docRequestAttachments.length > 0 ? docRequestAttachments[0] : undefined

      const { attachedDocuments, documentsByType } = this.props

      const attachedDocumentsForType = attachedDocuments.has(docType.id) ? attachedDocuments.get(docType.id) : []
      const documentsInLibraryForType = documentsByType.has(docType.id) ? documentsByType.get(docType.id) : []

      const attachedDocsData = attachedDocumentsForType.map(doc => {
        const relatedDoc = documentsInLibraryForType.find(d => d.id === doc.documentId)

        return {
          attachedDoc: doc,
          doc: relatedDoc
        }
      })

      // get registered library docs, which are not uploaded in this session
      const numberOfDocsInLibrary = documentsInLibraryForType.filter(
        doc =>
          doc.state === 'REGISTERED' &&
          !attachedDocumentsForType.find(attached => attached.source === 'upload' && attached.documentId === doc.id)
      ).length

      const pendingDoc = (
        attachedDocsData.find(d => d.attachedDoc.source === 'upload' && d.doc.state === 'PENDING') || ({} as any)
      ).doc

      const numberOfSelectedDocuments = attachedDocsData.filter(d => d.doc.state === 'REGISTERED').length

      return (
        <>
          {/* Has document attached? Show download / view button */
          !!requestAttachment ? (
            <DownloadButton
              docType={docType}
              document={requestAttachment}
              onOriginalDocument={this.props.onOriginalDocument}
              onDocumentDownload={this.props.onDocumentAttachmentDownload}
            />
          ) : (
            <span />
          )}
          <div>
            <AttachDocumentsDropdownWrap>
              <AttachDocumentsDropdown
                documentType={docType}
                automatchCount={numberOfDocsInLibrary}
                attachedDocument={pendingDoc}
                disabled={!this.canAttachDocForType(requestAttachment, docType.id)}
                toggleAddDocumentModalVisible={() => this.props.addNewDocumentRequested(docType)}
                toggleAutomatchModalVisible={() => this.props.automatchSelectRequested(docType)}
              />
            </AttachDocumentsDropdownWrap>
            {!!numberOfSelectedDocuments && (
              <ViewAllWrap onClick={() => this.handleViewDocuments(docType)}>
                View all ({numberOfSelectedDocuments})
              </ViewAllWrap>
            )}
          </div>
        </>
      )
    }
    return <TileWithItems key={`tile-${docType.id}`} documentType={docType} renderItems={renderTileItems} />
  }

  private getNumberOfRequestedDocumentTypes(): number {
    return this.props.documentRequest.types.length
  }

  private getNumberOfAlreadySentDocTypes(alreadySentDocs: Map<string, Document[]>): number {
    return alreadySentDocs.size
  }

  private handleDocDelete(activePreviewDocsIds: string[], docId: string) {
    // last doc delete requested
    if (activePreviewDocsIds.length === 1 && activePreviewDocsIds[0] === docId) {
      this.setState({
        activePreviewForType: null
      })
    }

    this.props.deleteDocRequested(docId)
  }
}

export const StyledTab = styled.div`
  display: flex;
  flex-wrap: wrap;
  margin-top: 8px;
  margin-bottom: 12px;
`

export const StyledListDocTypes = styled(List)`
  &&&&&&& {
    padding: 0px;
    margin-top: 0px;
    width: -webkit-fill-available;
    height: 280px;
    overflow-y: auto;
  }
`

const StyledTabItem =
  styled.span <
  { active: boolean } >
  `
  text-align: center;
  margin-right: 30px;
  cursor: pointer;
  color: ${blueGrey};
  font-weight: bold;
  padding-bottom: 6px;

  &:hover {
    color: black;
  }

  ${props =>
    props.active
      ? `
        color: ${violetBlue}
        border-bottom: 2px solid ${violetBlue};
        &:hover {
          color: ${violetBlue};
        }
      `
      : ``};

`

const CountWrap = styled.span`
  margin-left: 4px;
  font-weight: normal;
`

const StyledContent = styled.div`
  display: flex;
  flex-direction: column;
  min-width: 830px;
`

const AttachDocumentsDropdownWrap = styled.div`
  display: inline-block;
`

const ViewAllWrap = styled(SpanAsLink)`
  margin-left: ${SPACES.DEFAULT};
`
