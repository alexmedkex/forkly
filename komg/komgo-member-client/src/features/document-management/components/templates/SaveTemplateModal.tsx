import * as React from 'react'
import { Template } from '../../store'
import { Input, Modal } from 'semantic-ui-react'
import ModalFooterButton from '../ModalFooterButtons'

interface Props {
  visible: boolean
  template: Template
  toggleVisible(): void
  onSave(name: string): void
}

interface State {
  templateName: string
}

class SaveTemplateModal extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props)
    this.state = {
      templateName: ''
    }
  }

  handleChange = (e: any) => {
    this.setState({ templateName: e.target.value })
  }

  handleSave = () => {
    this.props.onSave(this.state.templateName)
    this.props.toggleVisible()
  }

  render() {
    return (
      <Modal
        width={'50%'}
        open={this.props.visible}
        centered={true}
        onOk={this.props.toggleVisible}
        onCancel={this.props.toggleVisible}
      >
        <Modal.Header>Save as template</Modal.Header>
        <Modal.Content>
          <p>Template name</p>
          <Input placeholder={'...'} onChange={this.handleChange} />
          <ModalFooterButton
            key="btn-back"
            toStep={0}
            text={'Cancel'}
            type={'default'}
            onClick={this.props.toggleVisible}
          />,
          <ModalFooterButton key="btn-next" toStep={1} text={'Save'} type={'primary'} onClick={this.handleSave} />
        </Modal.Content>
      </Modal>
    )
  }
}

export default SaveTemplateModal
