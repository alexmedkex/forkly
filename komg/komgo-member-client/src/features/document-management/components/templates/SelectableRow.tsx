import * as React from 'react'

import { Template } from '../../store'
import { Table } from 'semantic-ui-react'

interface RowProps {
  template: Template
  active: boolean
  rowId: any
  onClick(arg1: any, arg2: any): void
}

class SelectableRow extends React.Component<RowProps> {
  constructor(props: RowProps) {
    super(props)
  }

  onClick = (e: any) => {
    const { onClick, rowId } = this.props
    onClick(rowId, e)
  }

  render() {
    const { template, active } = this.props
    return (
      <Table.Row onClick={this.onClick} active={active}>
        {/* TODO: Find value for template type */}
        <Table.Cell>{template.name}</Table.Cell>
        <Table.Cell>{template.predefined ? 'System' : 'Custom'}</Table.Cell>
        <Table.Cell>
          {template.types.length} document{template.types.length > 1 ? 's' : ''}
        </Table.Cell>
      </Table.Row>
    )
  }
}

export default SelectableRow
