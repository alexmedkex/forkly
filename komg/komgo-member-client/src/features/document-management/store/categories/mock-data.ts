import { Category } from '../types'
import { mockProduct } from '../products/mock-data'

export const mockCategories: Category[] = [
  {
    id: 'company-details',
    product: mockProduct,
    name: 'Company Details'
  },
  {
    id: 'management-and-directors',
    product: mockProduct,
    name: 'Management and directors'
  },
  {
    id: 'shareholders',
    product: mockProduct,
    name: 'Shareholders - Ultimate Beneficiary Owners (UBOs)'
  },
  {
    id: 'business-description',
    product: mockProduct,
    name: 'Business description'
  },
  {
    id: 'regulation-and-compliance',
    product: mockProduct,
    name: 'Regulation/Compliance'
  },
  {
    id: 'banking-documents',
    product: mockProduct,
    name: 'Banking documents'
  },
  {
    id: 'miscellaneous',
    product: mockProduct,
    name: 'Miscellaneous'
  }
]
