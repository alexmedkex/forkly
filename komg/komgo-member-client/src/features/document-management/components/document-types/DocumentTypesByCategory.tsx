import * as React from 'react'
import { List, Accordion, AccordionPanelProps, Checkbox, Icon } from 'semantic-ui-react'
import styled from 'styled-components'
import { indigo, darkViolet } from '../../../../styles/colors'

import { DocumentType, Category } from '../../../document-management/store/types'
import { groupBy } from '../../../document-management/components/documents/my-documents/toMap'

import { PanelCheckboxState } from '../templates/DocumentRequestModal'

export interface Props {
  documentTypes: DocumentType[]
  categories: Category[]
  selectedDocumentTypesByCategory: Map<string, DocumentType[]>
  panelCheckboxState: Map<string, PanelCheckboxState>
  markAsDisabled: (documentTypeName: string) => { isDisabled: boolean; disabledText: string }
  isSelected(documentType: DocumentType): boolean
  onPanelClick(category: Category): void
  onListItemClick(documentType: DocumentType): void
}

class DocumentTypesByCategory extends React.Component<Props> {
  render() {
    return (
      <StyledAccordion
        exclusive={false}
        fluid={true}
        panels={this.props.categories.map(this.categoryToAccordionPanel)}
      />
    )
  }

  private categoryToAccordionPanel = (category: Category): AccordionPanelProps => {
    return {
      key: category.id,
      name: category.name,
      title: { content: this.categoryToPanelTitle(category) },
      content: { content: this.categoryToList(category) }
    }
  }

  private categoryToPanelTitle = (category: Category) => {
    return (
      <>
        <StyledIcon name="chevron down" />
        <StyledCheckbox
          key={category.id}
          floated="left"
          label={category.name}
          checked={this.props.panelCheckboxState.get(category.id) === 1}
          indeterminate={this.props.panelCheckboxState.get(category.id) === 0}
          onClick={(e: React.MouseEvent) => {
            e.preventDefault()
            e.stopPropagation()
            this.props.onPanelClick(category)
          }}
        />
      </>
    )
  }

  private categoryToList = (category: Category) => {
    const groupByCategoryId = (docType: DocumentType) => docType.category.id
    const documentTypesForThisCategory = groupBy(this.props.documentTypes, groupByCategoryId).get(category.id)
    if (!(documentTypesForThisCategory && documentTypesForThisCategory.length)) {
      return null
    }

    return (
      <List
        selection={true}
        name={<b>category.name</b>}
        items={documentTypesForThisCategory.map(this.documentTypeToListItem)}
      />
    )
  }

  private documentTypeToListItem = (documentType: DocumentType) => {
    // default values
    let disabledText: string = ''
    let isDisabled: boolean = false

    // override defaults if markAsDisabled function is provided
    if (this.props.markAsDisabled) {
      const result = this.props.markAsDisabled(documentType.name)
      if (result.isDisabled) {
        disabledText = result.disabledText
        isDisabled = result.isDisabled
      }
    }
    return (
      <List.Item style={{ padding: '7px 2.5em' }}>
        <List.Content key={documentType.id} floated="left">
          <Checkbox
            disabled={false} // We are allowing to re-request same document several times
            label={documentType.name}
            onClick={() => this.props.onListItemClick(documentType)}
            checked={this.props.isSelected(documentType)}
          />
        </List.Content>
        <List.Content key={documentType.id} floated="right">
          <p className="disabled" hidden={!isDisabled}>
            {disabledText}
          </p>
        </List.Content>
      </List.Item>
    )
  }
}

export const StyledAccordion = styled(Accordion)`
  &&& {
    .komgo-accordion-title {
      background-color: ${indigo} !important;
      border-radius: 5px;
      border: 1px solid ${darkViolet};
      padding: 12px 30px;
      height: 45px;
      color: white;
    }

    .title .dropdown.icon {
      display: none;
    }

    .active .komgo-accordion-title {
      border-radius: 5px 5px 0px 0px;
      margin-bottom: -14px;
    }

    i {
      color: white;
      margin-top: 2px;
    }
  }
`

const StyledCheckbox = styled(Checkbox)`
  &&& {
    padding-left: 1em;
    label {
      font-weight: bold;
    }
    .dropdown.icon {
      display: none;
    }
  }
`

const StyledIcon = styled(Icon)`
  &&& {
    width: 16px;
  }
`

export default DocumentTypesByCategory
