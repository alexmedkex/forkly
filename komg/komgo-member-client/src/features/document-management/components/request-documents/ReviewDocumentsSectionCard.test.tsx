import { shallow } from 'enzyme'
import * as React from 'react'

import { mockData } from '../../store/templates/mock-data'
import { Category, DocumentType, Request } from '../../store'
import { mockProduct } from '../../store/products/mock-data'
import * as renderer from 'react-test-renderer'
import { ReviewDocumentsSectionCard, Props } from './ReviewDocumentsSectionCard'
import { mockDocumentTypes } from '../../store/document-types/mock-data'

describe('IncomingRequestContainer component', () => {
  const MOCK_REQUEST_ID = '-999'

  const mockRequest: Request = {
    companyId: '-1',
    product: mockProduct,
    types: mockDocumentTypes,
    name: 'anon',
    id: MOCK_REQUEST_ID,
    sentDocuments: [],
    documents: [],
    notes: []
  }

  const mockFunc = jest.fn(() => void 0)

  const mockProps: Props = {
    documentRequest: mockRequest,
    onOriginalDocument: mockFunc,
    onDocumentAttachmentDownload: jest.fn(),
    documentsByType: new Map(),
    attachedDocuments: new Map(),
    automatchSelectRequested: jest.fn(),
    addNewDocumentRequested: jest.fn(),
    deleteDocRequested: jest.fn(),
    resetLoadedDocument: jest.fn(),
    downloadedRequestAttachmentForTypes: [],
    openViewDocument: jest.fn()
  }

  it('should render an empty DocumentTypeSelector item with props', () => {
    const wrapper = shallow(<ReviewDocumentsSectionCard {...mockProps} />)
    expect(wrapper.find('StyledTab').exists).toBeTruthy()
    expect(wrapper.find('StyledListDocTypes').exists).toBeTruthy()
    expect(wrapper.find('TileWithItems').exists).toBeTruthy()
  })

  it('renders', () => {
    expect(renderer.create(<ReviewDocumentsSectionCard {...mockProps} />).toJSON()).toMatchSnapshot()
  })
})
