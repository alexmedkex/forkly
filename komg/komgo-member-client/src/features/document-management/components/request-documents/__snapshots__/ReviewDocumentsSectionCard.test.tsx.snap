// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IncomingRequestContainer component renders 1`] = `
.c1.c1.c1.c1.c1.c1.c1.c1.c1.c1.c1 {
  margin: 0px;
  padding: 16px 30px 30px 30px;
}

.c0.c0.c0.c0.c0.c0.c0.c0.c0.c0.c0 {
  height: 400px;
  width: unset;
  margin: 30px;
  padding-left: 0px;
  padding-rigth: 0px;
  -webkit-box-shadow: 0px 1px 11px -4px rgba(0,0,0,0.75);
  -moz-box-shadow: 0px 1px 11px -4px rgba(0,0,0,0.75);
  box-shadow: 0px 1px 11px -4px rgba(0,0,0,0.75);
}

.c2.c2.c2.c2.c2.c2.c2.c2.c2.c2.c2 {
  font-size: 11px;
  height: 21px;
  width: 549px;
  color: #1c2936;
  font-weight: 600;
  line-height: 21px;
}

.c10 {
  height: 100%;
  min-width: 3.99px;
  -webkit-transform: scaleX(-1);
  -ms-transform: scaleX(-1);
  transform: scaleX(-1);
  background-color: rgba(63,81,181);
  border-radius: 0 4px 4px 0;
}

.c16 {
  height: 100%;
  min-width: 3.99px;
  -webkit-transform: scaleX(-1);
  -ms-transform: scaleX(-1);
  transform: scaleX(-1);
  background-color: rgba(0,179,188);
  border-radius: 0 4px 4px 0;
}

.c17 {
  height: 100%;
  min-width: 3.99px;
  -webkit-transform: scaleX(-1);
  -ms-transform: scaleX(-1);
  transform: scaleX(-1);
  background-color: rgba(68,33,133);
  border-radius: 0 4px 4px 0;
}

.c9 {
  height: 42px;
  width: calc(100% - 14px);
  margin: 4px 0;
  border: 1px solid #e8eef3;
  border-radius: 4px;
  display: grid;
  grid-template-columns: 5px [border] 1fr [type-name] 1fr [add-requirements-btn] 1fr [attach-document-btn] 28px [remove-icon];
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-column-gap: 0.5rem;
}

.c11 {
  color: #1c2936;
  font-size: 14px;
  font-weight: 600;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c13.c13.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 348px;
}

.c14.c14.c14 {
  color: #7f95aa;
  font-size: 14px;
}

.c15.c15.c15 {
  color: #5d768f;
  font-size: 13px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-top: 8px;
  margin-bottom: 12px;
}

.c8.c8.c8.c8.c8.c8.c8 {
  padding: 0px;
  margin-top: 0px;
  width: -webkit-fill-available;
  height: 280px;
  overflow-y: auto;
}

.c5 {
  text-align: center;
  margin-right: 30px;
  cursor: pointer;
  color: #5d768f;
  font-weight: bold;
  padding-bottom: 6px;
  color: #5700b5;
  border-bottom: 2px solid #5700b5;
}

.c5:hover {
  color: black;
}

.c5:hover {
  color: #5700b5;
}

.c7 {
  text-align: center;
  margin-right: 30px;
  cursor: pointer;
  color: #5d768f;
  font-weight: bold;
  padding-bottom: 6px;
}

.c7:hover {
  color: black;
}

.c6 {
  margin-left: 4px;
  font-weight: normal;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-width: 830px;
}

.c12 {
  display: inline-block;
}

<div
  className="ui card c0"
  height="400px"
  onClick={[Function]}
  style={Object {}}
>
  <div
    className="content c1"
  >
    <div
      className="header c2"
    >
      REQUEST DETAILS
    </div>
    <div
      className="description "
    >
      <div
        className="c3"
      >
        <div
          className="c4"
        >
          <span
            className="c5"
            onClick={[Function]}
          >
            Requested 
             
            <span
              className="c6"
            >
              (
              3
              )
            </span>
          </span>
          <span
            className="c7"
            onClick={[Function]}
          >
            Already sent 
             
            <span
              className="c6"
            >
              (
              0
              )
            </span>
          </span>
        </div>
      </div>
      <div
        className="ui list style-scroll c8"
        role="list"
      >
        <div
          className="c9"
          data-test-id="requirements-tile-Certificate of Incorporation"
        >
          <div
            className="c10"
          />
          <div
            className="c11"
            data-test-id="requirement-field-name-Certificate of Incorporation"
          >
            Certificate of Incorporation
          </div>
          <span />
          <div>
            <div
              className="c12"
            >
              <div
                aria-disabled={false}
                aria-expanded={false}
                className="ui button fluid dropdown"
                data-test-id="attach-documents-button-Certificate of Incorporation"
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseDown={[Function]}
                role="listbox"
                style={
                  Object {
                    "width": "238px",
                  }
                }
                tabIndex={0}
              >
                <div
                  aria-live="polite"
                  className="text"
                  role="alert"
                >
                  Attach document
                </div>
                <i
                  aria-hidden="true"
                  className="dropdown icon"
                  onClick={[Function]}
                />
                <div
                  className="menu transition"
                >
                  <div
                    aria-disabled={true}
                    className="disabled item"
                    data-test-id="automatch-button-Certificate of Incorporation"
                    onClick={[Function]}
                    role="option"
                  >
                    <div
                      className="text c13"
                    >
                      <span
                        className="c14"
                      >
                        Select from library
                      </span>
                      <span
                        className="c15"
                      >
                        [ 0 automatch ]
                      </span>
                    </div>
                  </div>
                  <div
                    className="item"
                    data-test-id="add-document-button=Certificate of Incorporation"
                    onClick={[Function]}
                    role="option"
                  >
                    <span
                      className="text"
                    >
                      Upload new
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="c9"
          data-test-id="requirements-tile-List of Directors"
        >
          <div
            className="c16"
          />
          <div
            className="c11"
            data-test-id="requirement-field-name-List of Directors"
          >
            List of Directors
          </div>
          <span />
          <div>
            <div
              className="c12"
            >
              <div
                aria-disabled={false}
                aria-expanded={false}
                className="ui button fluid dropdown"
                data-test-id="attach-documents-button-List of Directors"
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseDown={[Function]}
                role="listbox"
                style={
                  Object {
                    "width": "238px",
                  }
                }
                tabIndex={0}
              >
                <div
                  aria-live="polite"
                  className="text"
                  role="alert"
                >
                  Attach document
                </div>
                <i
                  aria-hidden="true"
                  className="dropdown icon"
                  onClick={[Function]}
                />
                <div
                  className="menu transition"
                >
                  <div
                    aria-disabled={true}
                    className="disabled item"
                    data-test-id="automatch-button-List of Directors"
                    onClick={[Function]}
                    role="option"
                  >
                    <div
                      className="text c13"
                    >
                      <span
                        className="c14"
                      >
                        Select from library
                      </span>
                      <span
                        className="c15"
                      >
                        [ 0 automatch ]
                      </span>
                    </div>
                  </div>
                  <div
                    className="item"
                    data-test-id="add-document-button=List of Directors"
                    onClick={[Function]}
                    role="option"
                  >
                    <span
                      className="text"
                    >
                      Upload new
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="c9"
          data-test-id="requirements-tile-Passports of UBOs"
        >
          <div
            className="c17"
          />
          <div
            className="c11"
            data-test-id="requirement-field-name-Passports of UBOs"
          >
            Passports of UBOs
          </div>
          <span />
          <div>
            <div
              className="c12"
            >
              <div
                aria-disabled={false}
                aria-expanded={false}
                className="ui button fluid dropdown"
                data-test-id="attach-documents-button-Passports of UBOs"
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseDown={[Function]}
                role="listbox"
                style={
                  Object {
                    "width": "238px",
                  }
                }
                tabIndex={0}
              >
                <div
                  aria-live="polite"
                  className="text"
                  role="alert"
                >
                  Attach document
                </div>
                <i
                  aria-hidden="true"
                  className="dropdown icon"
                  onClick={[Function]}
                />
                <div
                  className="menu transition"
                >
                  <div
                    aria-disabled={true}
                    className="disabled item"
                    data-test-id="automatch-button-Passports of UBOs"
                    onClick={[Function]}
                    role="option"
                  >
                    <div
                      className="text c13"
                    >
                      <span
                        className="c14"
                      >
                        Select from library
                      </span>
                      <span
                        className="c15"
                      >
                        [ 0 automatch ]
                      </span>
                    </div>
                  </div>
                  <div
                    className="item"
                    data-test-id="add-document-button=Passports of UBOs"
                    onClick={[Function]}
                    role="option"
                  >
                    <span
                      className="text"
                    >
                      Upload new
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
